### PR TITLE
Revert "LPS-64106 Each module's upgrade process must disable the cache"

### PR DESCRIPTION
**Pull request by shuyangzhou**
This reverts commit 2889e23eb2bcdb5b50ecc90a972d2228b95a2069.

@brianchandotcom @mhan810 this is the fix to the random BackgroundTask MVCC error in tomcat log scanning.

Basically we refactored the upgrade process, which made it no longer synchronously runs at the very beginning. As a result some of the upgrade logic for modules run concurrently with the rest of startup logic.

In this MVCC error case, modules upgrade logic runs together with Background tasks for startup indexing (on MB threads). And the upgrade logic for each module does this cache disable-enable thing due to 2889e23eb2bcdb5b50ecc90a972d2228b95a2069. As a result, we have case like this:
1) At time point 1, the cache is on. Background task saves something into the cache.
2) At time point 2, Background task modifies that entry, and does a persistence update. But the cache is off at this moment. So the update goes into DB, but not cache.
3) At time point 3, Background rereads that entry, and does another modification and update. And the cache is on, it reads the out of date result from 1), then proceed the modification and update, then rejected by hibernate due to an out of date MVCC version.

Basically we have changed the natural of the upgrade, it is no longer a startup only serialized thing, it can run concurrently with the rest of portal logic, and more importantly, if someone deploys a module with upgrade logic, after portal startup. There is a very good chance we blow off some runtime real business logic.

@mdelapenya on the ticket and commit, I don't see why it was added. For whatever reason the cache was disabled, we need to fix it in a different way. We can not flip the cache switch at runtime.

CC @willnewbury

### CODE REVIEW

**Comment by shuyangzhou** [comment]
I talked to @brianchandotcom, appended a new commit to clear the cache in the end to make sure everyone else can see the upgrade changes. This is very heavy, but should work, after all upgrade does not run very often.

The difference between fliping the cache switch and clearing cache in the end is, fliping the cache switch can cause out of date read system wide. Clearing cache could only possibly causes issues to the caches for the upgrading tables. The upgrade logic should ensure when it runs, no one else is reading from that table, therefore it should be safe. Otherwise we will have some upgrade code to fix.

**Comment by shuyangzhou** [comment]
ci:retest

**Comment by brianchandotcom** [comment]
Merged. Thank you.
View total diff: https://github.com/brianchandotcom/liferay-portal/compare/3c85e47433...e21d1c5771

**Comment by brianchandotcom** [comment]
@mdelapenya fyi

**Comment by mdelapenya** [comment]
Cc/ @migue
